docs/development.rst

```diff
@@ -152,26 +152,107 @@ The top-level document to open with a web browser will be
+are used, or the installed package.
+
+The test case files and any utility functions they use are always used from
+the main repository directory, and are located in the `testsuite` directory
+structure:
 
 ::
 
-    $ make test                                              # Run all unit tests
-    $ PYTHONPATH=. py.test testsuite/test_cim_obj.py -s      # Run only this test source file
-    $ PYTHONPATH=. py.test InitCIMInstanceName -s            # Run only this test class
-    $ PYTHONPATH=. py.test -k InitCIMInstanceName or Bla -s  # py.test -k expressions are possible
+    testsuite
+     +-- unittest            Unit tests for single modules of the package
+     |    +-- pywbem              Unit tests for the pywbem package
+     |    +-- pywbem_mock         Unit tests for the pywbem_mock package
+     |    +-- unittest_utils      Unit tests for testsuite/unittest/utils
+     |    +-- functiontest        Unit tests for testsuite/functiontest
+     |    +-- end2endtest_utils   Unit tests for testsuite/end2endtest/utils
+     |    +-- servers             Unit tests for testsuite/servers
+     |    +-- utils               Utility functions for use by unit tests
+     +-- functiontest        Function tests for the entire pywbem client
+     +-- end2endtest         End2end tests against WBEM servers
+     |    +-- utils               Utility functions for use by end2end tests
+     +-- manualtest          Manual tests
+     +-- servers             WBEM server definition file used by some tests and module for accessing it
+     +-- profiles            Simple definitions of management profiles used by some tests
+     +-- schema              The CIM schema MOF files used by some MOF tests
+     +-- dtd                 The CIM DTD file used by some CIM-XML validation tests
+
+There are multiple types of tests in pywbem:
+
+1. Unit tests and function tests
+
+   These tests do not require any WBEM server to be available, and the tests
+   validate their results automatically.
+
+   They are run by executing:
+
+   ::
+
+       $ make test
+
+   Test execution can be modified by a number of environment variables, as
+   documented in the make help (issue `make help`).
+
+2. End2end tests
+
+   These tests are run against one or more WBEM servers, and the tests validate
+   their results automatically.
+
+   They are run by preparing a server definition file:
+
+   ::
+
+       testsuite/end2endtest/server_file.yaml
+
+   from the provided example, and by executing:
+
+   ::
+
+       $ make end2end
+
+   Again, test execution can be modified by a number of environment variables,
+   as documented in the make help (issue `make help`).
+
+3. Manual tests
+
+   There are several Python scripts and shell scripts that can be run manually.
+   The results need to be validated manually.
+
+   These scripts are in the directory:
+
+   ::
+
+       testsuite/manualtest
+
+   and are executed by changing into that directory and executing the
```

Comment:
Do you really have to change into that directory?

Reply:
I have looked at some of the scripts and it seems at least one has a dependency on running with the main directory being the current directory. So I'll change that accordingly.

Reply:
DONE